drivers/volume/csi/csi.go

```diff
@@ -0,0 +1,1023 @@
+package csi
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	kSnapshotv1beta1 "github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
+	kSnapshotClient "github.com/kubernetes-csi/external-snapshotter/v2/pkg/client/clientset/versioned"
+	"gocloud.dev/gcerrors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	snapv1 "github.com/kubernetes-incubator/external-storage/snapshot/pkg/apis/crd/v1"
+	snapshotVolume "github.com/kubernetes-incubator/external-storage/snapshot/pkg/volume"
+	storkvolume "github.com/libopenstorage/stork/drivers/volume"
+	storkapi "github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	"github.com/libopenstorage/stork/pkg/crypto"
+	"github.com/libopenstorage/stork/pkg/errors"
+	"github.com/libopenstorage/stork/pkg/log"
+	"github.com/libopenstorage/stork/pkg/objectstore"
+	"github.com/portworx/sched-ops/k8s/core"
+	storkops "github.com/portworx/sched-ops/k8s/stork"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	// storkDriverName is the name of the k8s driver implementation.
+	// not to be confused with a CSI Driver Name
+	storkCSIDriverName = "csi"
+	// snapshotPrefix is appended to CSI backup snapshot
+	snapshotPrefix = "snapshot-"
+	// snapshotClassNamePrefix is the prefix for snapshot classes per CSI driver
+	snapshotClassNamePrefix = "stork-csi-snapshot-class-"
+
+	// snapshotObjectName is the object stored for the volumesnapshot
+	snapshotObjectName = "snapshots.json"
+	// snapshotObjectName is the object stored for the volumesnapshot
+	storageClassesObjectName = "storageclasses.json"
```

Comment:
The SC is required for restoring. Since we're creating a new PVC and PV, we need to reference the original SC. The other option is to specificy a new SC upon restore. Is that preferred? 

Is there a way to specific a snapshotclass in `ApplicationBackupSpec`, or should I add as an Option?

Reply:
It can be a key/value under `Option`